pyteal/ast/subroutine_test.py

```diff
@@ -243,72 +392,21 @@ def one_dynscratchvar_impl(x: pt.DynamicScratchVar):
     )
 
     # Now we're back to validate() and everything should be copacetic
-
-    # input type handling:
     for x, y, z in product(pt.TealType, pt.TealType, pt.TealType):
-        params, anns, arg_types, byrefs, abis = three_params._validate(
-            input_types=[x, y, z]
-        )
+        (
+            params,
+            anns,
+            arg_types,
+            byrefs,
+            abi_args,
+            output_kwarg,
+        ) = three_params._validate(input_types=[x, y, z])
         assert len(params) == 3
         assert anns == {}
         assert all(at is pt.Expr for at in arg_types)
         assert byrefs == set()
-        assert abis == {}
-
-    # annotation / abi type handling:
-    abi_annotation_examples = {
```

Comment:
These cases are no longer necessary, or they don't make sense anymore? From my experience, erroring on the side of redundancy is preferred over under-coverage.

Reply:
But if keeping these cases made the merge too difficult, and there is comparable coverage elsewhere, then it makes sense to drop them.

Reply:
let me look into this... PRs are clobbing each others so I might over-deleted something 🤦

Reply:
Thanks for bringing them back, I hope it wasn't too much pain to do so.  I will probably need to upgrade these in my next PR, but these look good for now. I will approve as all my questions have been addressed.